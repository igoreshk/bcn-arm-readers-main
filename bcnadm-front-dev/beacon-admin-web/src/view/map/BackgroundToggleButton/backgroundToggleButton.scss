.activeMapButtonIcon {
  width: 24px;
  height: 24px;
  fill: #ffffff;
}

.inactiveMapButtonIcon {
  width: 24px;
  height: 24px;
  fill: #39c1d7;
}
