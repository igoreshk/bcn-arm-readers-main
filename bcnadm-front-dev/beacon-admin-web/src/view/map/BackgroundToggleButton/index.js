import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import Map from '@material-ui/icons/Map';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { toggleBackground } from 'src/reducers/mapBackgroundReducer/mapBackgroundSlice';

import ModeButton, { buttonType } from '../ModeButton';

import './backgroundToggleButton.scss';

class BackgroundToggleButton extends Component {
  componentDidMount() {
    const { showMapBackground } = this.props;

    // If current state of showMapBackground is false, toggle it
    if (showMapBackground !== true) {
      this.props.toggleBackground();
    }
  }

  handleClick = () => {
    this.props.toggleBackground();
  };

  render() {
    const { showMapBackground } = this.props;
    const activeClassName = showMapBackground ? 'activeMapButtonIcon' : 'inactiveMapButtonIcon';

    return (
      <>
        <ModeButton
          icon={<Map className={activeClassName} />}
          onClick={this.handleClick}
          active={showMapBackground}
          buttonId="toggleBackgroundMapButton"
          buttonType={buttonType.MODE}
        />
      </>
    );
  }
}

BackgroundToggleButton.propTypes = {
  showMapBackground: PropTypes.bool.isRequired,
  toggleBackground: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  showMapBackground: state.mapBackground.showMapBackground
});

const mapDispatchToProps = {
  toggleBackground
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(BackgroundToggleButton));
