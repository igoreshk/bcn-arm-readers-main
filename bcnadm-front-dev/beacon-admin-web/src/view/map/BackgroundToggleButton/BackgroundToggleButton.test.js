import React from 'react';
import { shallow } from 'enzyme';

import BackgroundToggleButton from 'src/view/map/BackgroundToggleButton/index';

const props = {
  showMapBackground: true,
  toggleBackground: jest.fn()
};

const wrapper = shallow(<BackgroundToggleButton {...props} />);

describe('mock', () => {
  it('render component', () => expect(wrapper.length).toEqual(1));
  it('snapshot for BackgroundToggleButton', () => expect(wrapper).toMatchSnapshot());
});
