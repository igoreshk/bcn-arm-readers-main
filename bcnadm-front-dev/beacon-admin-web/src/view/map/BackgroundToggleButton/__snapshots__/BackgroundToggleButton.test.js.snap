// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mock snapshot for BackgroundToggleButton 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
